docs/docs/troubleshooting.md

```diff
@@ -0,0 +1,165 @@
+
+To clarify; this does not mean that the upstream service is rejecting the JWT singing key. Rather, it doesn't trust the Let's Encrypt certificate used by the Authorize service for TLS, and so it will not read the JWKS file.
+
+For upstream applications that can use a local signing key file, you can circumvent this issue using `curl` or `wget` to download the signing key locally (relative to the upstream service). Using Grafana again as an example:
+
+1. Download the `jwks.json` file from the authenticate server:
+
+    ::::: tabs
+    :::: tab curl
+
+    ```bash
+    curl https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json > /etc/grafana/jwks.json
+    ```
+    ::::
+    :::: tab wget
+
+    ```bash
+    wget -O /etc/grafana/jwks.json https://authenticate.localhost.pomerium.io/.well-known/pomerium/jwks.json
+    ```
+    ::::
+    :::::
+
+1. Edit the upstream service configuration to use the local key to verify tokens:
+
+    ```ini
+    [auth.jwt]
+    enabled = true
+    header_name = X-Pomerium-Jwt-Assertion
+    email_claim = email
+    jwk_set_file = /etc/grafana/jwks.json
+    cache_ttl = 60m
+    ```
+
+### Kubernetes Ingress Controller
+
+#### View Event History
+
+Pomerium Ingress Controller will add **events** to the Ingress objects as it processes them.
+
+```
+kubectl describe ingress/my-ingress
+```
+
+```log
+Events:
+  Type    Reason   Age   From              Message
+  ----    ------   ----  ----              -------
+  Normal  Updated  18s   pomerium-ingress  updated pomerium configuration
+```
+
+If an error occurs, it may be reflected in the events:
+
+```log
+Events:
+  Type     Reason       Age                 From              Message
+  ----     ------       ----                ----              -------
+  Normal   Updated      5m53s               pomerium-ingress  updated pomerium configuration
+  Warning  UpdateError  3s                  pomerium-ingress  upsert routes: parsing ingress: annotations: applying policy annotations: parsing policy: invalid rules in policy: unsupported conditional "maybe", only and, or, not, nor and action are allowed
+```
+
+### Redirect Loop with Redis Databroker
+
+When using Redis, the [shared secret](/reference/readme.md#shared-secret) is used to encrypt data in Redis itself. if you change the configured shared secret, data from Redis can no longer be decrypted. This results in errant behavior, including redirect loops when a user session cannot be retrieved from the databroker.
+
+The resolution is to flush the Redis database with [`FLUSHDB`](https://redis.io/commands/flushdb) or [`FLUSHALL`](https://redis.io/commands/FLUSHALL).
+
+### RPC Errors
+
+#### certificate signed by unknown authority
```

Comment:
Should this all be in lower case? Not sure, just pointing it out

Reply:
Yes, because in this case the header is a direct copy of an error message. This should only occur in H4s, and when it does we copy the case of the error message for optimum search.